### PR TITLE
Fix FS mocking

### PR DESCRIPTION
**Pull request by meyer**
Hi there @zimme. I was digging around my `node_modules` and saw that `prettier-eslint` was pulling in some old deps, and the project itself needs some maintenance. I’d like to help where I can. Figured I’d start with #202.

I do some virtual filesystem stuff in a different project of mine using proxies. Working code is below. After this change there’s just one test that fails:

Screenshot 2019-05-14 20 02 16

I’m not sure what node versions you support with this project, but argument destructuring and proxies are pretty well supported these days (back to node 6 i believe).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   "maintainers": [
     "Simon Fridlund <simon@fridlund.email>"
   ],
+  "prettier": {},
```
Comment (meyer) [inline package.json]: this resets prettier for folks who have global settings defined (like i do 😀️)

**Comment by zimme** [comment]
Awesome, I'll look at this some evening this weekend. This has been a blocker for accepting other PRs, so again, amazeballs!

As for the node versions, I don't think anyone can expect us to keep backwards compatibility further back then the LTS versions of node so I'm fine with us requiring node 8+ and dropping official support for node 8 at the end of the year when that hits end-of-life.

**Comment by meyer** [comment]
@zimme rad, thanks! No rush at all, this is still a WIP. Also something is funky with CI, not sure what’s up. When I get a sec I’ll try replicate the issue locally.

**Comment by zimme** [comment]
Closing in favor of #211.

Thanks for the work on this and let's consider adding the mocking from this PR into the package at a later time as a separate PR as it might be a better solution.

**Comment by meyer** [comment]
yay, glad it’s fixed!

**Pull request closed without merging**